docs/spec/v1alpha1/gitrepositories.md

```diff
@@ -0,0 +1,216 @@
+# Git Repositories
+
+The `GitReposiory` API defines a source for artifacts coming from Git. 
+
+## Specification
+
+Git repository spec:
+
+```go
+// GitRepositorySpec defines the desired state of GitRepository
+type GitRepositorySpec struct {
+	// +kubebuilder:validation:Pattern="^(http|https|ssh)://"
```

Comment:
I would move this line below the comment on the `URL` field so they stick together.